hax/hax/motr/__init__.py

```diff
@@ -77,7 +78,12 @@ def start(self, rpc_endpoint: str, process: Fid, ha_service: Fid,
 
         @repeat_if_fails()
         def _get_rm_fid() -> Fid:
-            return self.consul_util.get_rm_fid()
+            while (True):
+                try:
+                    return self.consul_util.get_rm_fid()
```

Comment:
I think the fix should have been simpler. 

Per my understanding, `self.consul_util.get_rm_fid()` raises something like `KeyError` if the leader is not elected. In that case the proper fix is to translate that exception into `HaConsistencyException` so that `@repeat_if_fails` will repeat the attempt for you.

What exception is thrown that causes hax to stop? In other words, how does the issue looks like?

Reply:
@knekrasov  Please see this [comment](https://jts.seagate.com/browse/EOS-25926?focusedCommentId=2237688&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-2237688) for issue

Reply:
Ok, I'll add the stacktrace for clarity:
```
2021-11-11 17:59:29,393 [DEBUG] {qconsumer-31} Waiting for the next message
2021-11-11 17:59:29,393 [DEBUG] {MainThread} Starting m0_halon_interface
2021-11-11 17:59:29,394 [DEBUG] {MainThread} KVGET key=leader, kwargs={}
2021-11-11 17:59:30,435 [ERROR] {MainThread} Exiting due to an exception
Traceback (most recent call last):
  File "/opt/seagate/cortx/hare/lib64/python3.6/site-packages/hax/hax.py", line 167, in main
    profile=cfg.profiles[0])
  File "/opt/seagate/cortx/hare/lib64/python3.6/site-packages/hax/motr/__init__.py", line 82, in start
    rm_fid = _get_rm_fid()
  File "/opt/seagate/cortx/hare/lib64/python3.6/site-packages/hax/util.py", line 143, in wrapper
    return f(*args, **kwds)
  File "/opt/seagate/cortx/hare/lib64/python3.6/site-packages/hax/motr/__init__.py", line 80, in _get_rm_fid
    return self.consul_util.get_rm_fid()
  File "/opt/seagate/cortx/hare/lib64/python3.6/site-packages/hax/consul/cache.py", line 126, in wrapper
    ret_value = f(*args, **kwds)
  File "/opt/seagate/cortx/hare/lib64/python3.6/site-packages/hax/util.py", line 368, in get_rm_fid
    rm_node = self.get_session_node(self.get_leader_session())
  File "/opt/seagate/cortx/hare/lib64/python3.6/site-packages/hax/util.py", line 451, in get_session_node
    return str(session['Node'])  # principal RM
TypeError: 'NoneType' object is not subscriptable
2021-11-11 17:59:30,437 [INFO] {MainThread} Finalizing motr
```

So the core problem is that `get_session_node` doesn't expect that `session` object is None. 
## Suggestion: 
1. Modify `get_session_node` method so that it raises `HAConsistencyException` if `session` is None.

No other changes are required.